static/app/utils/usePrismTokens.tsx

```diff
@@ -0,0 +1,151 @@
+import {useEffect, useMemo, useState} from 'react';
+import * as Sentry from '@sentry/react';
+import * as Prism from 'prismjs';
+
+import {loadPrismLanguage, prismLanguageMap} from 'sentry/utils/loadPrismLanguage';
+
+type PrismHighlightParams = {
+  code: string;
+  language: string;
+};
+
+export type SyntaxHighlightToken = {
+  children: string;
+  className: string;
+};
+
+export type SyntaxHighlightLine = SyntaxHighlightToken[];
+
+type IntermediateToken = {
+  children: string;
+  types: Set<string>;
+};
+
+const useLoadPrismLanguage = (language: string, {onLoad}: {onLoad: () => void}) => {
+  useEffect(() => {
+    if (!language) {
+      return;
+    }
+
+    loadPrismLanguage(language, {
+      onLoad,
+      onError: () => {
+        Sentry.withScope(scope => {
+          scope.setTag('prism_language', language);
+          Sentry.captureException(
+            new Error('Prism.js failed to load language for stack trace')
+          );
```

Comment:
I don't think this needs its own tag since we probably won't search for it. extra should show up in additional data
```suggestion
          Sentry.captureException(
            new Error('Prism.js failed to load language for stack trace'),
            extra: { language },
          );
```